right/src/config_parser/basic_types.c

```diff
@@ -26,6 +27,16 @@ uint32_t ReadUInt32(config_buffer_t *buffer)
     return uInt32;
 }
 
+float ReadFloat(config_buffer_t *buffer)
```

Comment:
Do we want some specific float format here?

Reply:
I think simply using a float should be fine. Why not use type casting instead of `memcpy()`? I'm unsure about endianness.

Reply:
`return *(float*)&res;` throws a warning... You are welcome to suggest a better way to cast it.

Reply:
`return (float)uInt32;` doesn't throw a warning for me.

Reply:
Yeah, but it also does not do what we want, since it _converts_ the integer into a float, instead of just casting it 😭.

Reply:
Unless you have another preference, I will change it to:
```
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wstrict-aliasing"
    return *((float*)&uInt32);
#pragma GCC diagnostic pop
```
which disables the warning...